### PR TITLE
feat: AA-1127: Adds ability to override Braze frequency caps when sen…

### PR DESCRIPTION
**Pull request by Dillon-Dumesnil**
…ding emails

**Description:** Braze API based campaigns are still subject to the global capping
rules defined in Braze. In order to get around this, you can specifically state
the email will override those rules. Now edx-ace supports that capability.

**JIRA:** [Link to JIRA ticket](https://openedx.atlassian.net/browse/AA-1127)

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPI after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.

### CODE REVIEW

File: edx_ace/channel/braze.py [hunk 1]
```diff
@@ -122,6 +122,7 @@ def deliver(self, message, rendered_message):
             return
 
         transactional = message.options.get('transactional', False)
+        override_frequency_capping = message.options.get('override_frequency_capping', False)
```
Comment (mikix) [inline edx_ace/channel/braze.py]: Is there an opportunity to just silently determine this setting? Like, is this the sort of thing we should just automatically do for transactional emails and not do for campaign emails?

Comment (Dillon-Dumesnil) [reply edx_ace/channel/braze.py]: I'm not sure honestly. I was thinking about just using the `transactional` field to set this value, but the emails we are currently focused on (dynamic pacing) are not marked as transactional and I don't think they are supposed to be either. So it's a bit hard to have a hardfast rule for this. Ideally, honestly, what we should have is the ability in Braze to start creating Transactional email campaigns which automatically ignore frequency caps, but alas, we don't have access to those still and have API triggered campaigns instead.

This did give me an idea though of changing this line to be 
```
override_frequency_capping = message.options.get('override_frequency_capping', transactional or False)
```
I think that would at least include all transactional emails then and then any other campaigns that explicitly call it out.

Comment (mikix) [reply edx_ace/channel/braze.py]: I like that. Can drop the `or False` though, for even simpler code.

Comment (mikix) [reply edx_ace/channel/braze.py]: (Normally, transactional emails don't even go through the braze channel - but they can be configured to do so by adding the message id to the BRAZE_CAMPAIGN setting. In which case, we almost certainly don't want the email to be frequency capped. So this change makes sense)

Comment (Dillon-Dumesnil) [reply edx_ace/channel/braze.py]: Yeah, agreed. And duh about the `or False`. My b

**Review by MatthewPiatetsky — COMMENTED** [review]
Not a blocker but wonder if it would be worthwhile to make a ticket to review our other campaigns and whether they need this setting?

**Comment by Dillon-Dumesnil** [comment]
> Not a blocker but wonder if it would be worthwhile to make a ticket to review our other campaigns and whether they need this setting?

https://openedx.atlassian.net/browse/AA-1129

**Review by mikix — APPROVED** [review]

**Pull request merged**